### DISCUSSION TITLE
Cogs object list can't be used in 'await' expression

### DISCUSSION THREAD

**Discussion by MelmanC** [Rapptz/discord.py#9239] [Q&A]
Hi! Sorry to disturb but I don't understand where the error comes from, if someone can help me?

![code](https://user-images.githubusercontent.com/25880339/219019012-1560413a-31e9-4dbf-adfe-da01f20935d0.png)
![code1](https://user-images.githubusercontent.com/25880339/219018313-6ce61fc0-8d04-48da-89a4-3e7ee0701b82.png)

**i have this error :**

C:\Users\nonod\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\discord\cog.py:786: RuntimeWarning: coroutine 'setup' was never awaited
  setup(self)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Traceback (most recent call last):
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 37, in <module>
    asyncio.run(main())
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 647, in run_until_complete
    return future.result()
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 33, in main
    await load()
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 29, in load
    await bot.load_extension(f'cogs.{filename[:-3]}')
TypeError: object list can't be used in 'await' expression

**Comment by LostLuma** [comment]
Hello, seeing that the line numbers from the code and your error do not match up your program has been modified since the screenshots have been taken, making it impossible to properly help you with your issue.

Additionally if you'd like to make the work of helpers easier post your code and errors inside code blocks instead of screenshots or raw text.

You can create code blocks on GitHub like this:

\`\`\`python
\# Paste your code here
\`\`\`

Which will turn into a more readable, and most importantly, easy to copy and test form:

```python
# Paste your code here
```

**Reply by MelmanC** [reply]
Ho sorry ! 

```python
import discord
from discord.ext import commands, tasks
from itertools import cycle
import asyncio
import os

bot = commands.Bot(command_prefix='/', intents=discord.Intents.all())

bot_status = cycle(["Author : test", "/help", "Status 3"])


@tasks.loop(seconds=5)
async def change_status():
    await bot.change_presence(activity=discord.Game(next(bot_status)))

    
@bot.event
async def on_ready():
    print(f'Bonjour, je suis {bot.user.name} et je suis connecté !')
    change_status.start()


async def load():
    for filename in os.listdir('./cogs'):
        if filename.endswith('.py'):
            await bot.load_extension(f'cogs.{filename[:-3]}')
            
async def main():
    await load()
    await bot.start('TOKEN')


asyncio.run(main())
```

and my ping.py

```python
import discord
from discord.ext import commands

class Ping(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.listener()
    async def on_ready(self):
        print('Ping loaded')


    @commands.command()
    async def ping(self, ctx):
        await ctx.send(f'pong with {round(self.bot.latency * 1000)} ms {ctx.author.mention}')

async def setup(bot):
    await bot.add_cog(Ping(bot))
```
so this is my error 

C:\Users\nonod\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\discord\cog.py:786: RuntimeWarning: coroutine 'setup' was never awaited
  setup(self)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Traceback (most recent call last):
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 33, in <module>
    asyncio.run(main())
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.3568.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 647, in run_until_complete
    return future.result()
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 29, in main
    await load()
  File "c:\Users\nonod\OneDrive\Bureau\Dev\Bot\main.py", line 26, in load
    await bot.load_extension(f'cogs.{filename[:-3]}')
TypeError: object list can't be used in 'await' expression

**Reply by LostLuma** [reply]
Thanks for the additional info.
Unfortunately I am not sure what causes your error and can not reproduce it. Let's hope someone else might be able to help!

**Reply by MelmanC** [reply]
Okay thx !

**Reply by LeoCx1000** [reply]
Could you perhaps try reinstalling the library? I can't reproduce it either.

``pip install --force-reinstall -U discord.py``

**Reply by MelmanC** [reply]
It's good , tysm !

**Reply by ElBe-Plaq** [reply]
Maybe try `async def __init__(self, bot):` instead of `def __init__(self, bot):` in line 5 of `ping.py`.

**Reply by AbstractUmbra** [reply]
Absolutely do not do this.

**Reply by ElBe-Plaq** [reply]
Why?

**Reply by AbstractUmbra** [reply]
Firstly just adding `await` isn't magical. It wont just work, there are things to consider when you make a function a coroutine function.

Secondly, `__init__` is a special dunder method that is called to initialise a class.
It is not async, making it an async def will error.

**Reply by ElBe-Plaq** [reply]
Okay, my bad.

**Reply by ElBe-Plaq** [reply]
It's a old version of python, maybe try installing `3.11`.
